### ISSUE TITLE
[gatsby-image-sharp]: maxWidth based on a 1x or 2x display?

### ISSUE THREAD

**Issue by ooloth** [gatsbyjs/gatsby#2849]
Quick question!

When I create an imageSharp query, is the `maxWidth` in the query based on a 1x or 2x pixel-density display?

In other words, do I need to double the width in the query?

For example, for a 1500px image, do I set a maxWidth of 1500px or 3000px?

```
imageQuery: allImageSharp(
  filter: { id: { regex: "/image/" } }
  sort: { fields: [id], order: ASC }
) {
  edges {
    node {
      sizes(maxWidth: 1500? 3000?) {
	 ...GatsbyImageSharpSizes
      }
    }
  }
}
```

**Comment by Undistraction** [comment]
You need to set the width of the largest image you will need, so if if your image has a `max-width` of 1500px or is constrained to a maximum of 1500px in some other way,  you will need an image that is 3000px for screens with a x2 pixel density, so your `maxWidth` setting will need to be `3000`.

**Comment by KyleAMathews** [comment]
tl;dr set the maxWidth to the actual maxWidth.

There are two ways images can be responsive. One is if you know the size of the image and the other is you don't.

When you know the size of the image, you query using `resolutions` as Gatsby then generates image thumbnails at different resolutions (1x, 1.5x, 2x, 3x).

If you don't know the size, then you say what the maxWidth is and then Gatsby will generate thumbnails at 2x & 3x the that width plugin 1/2, 1/3, 1/4 (can't remember exactly all of them). Then we tell the browser the width of each thumbnail and the browser takes care of picking the right image based on what it knows the width of the page to be and the resolution of the screen.

This is a good doc to read up on this https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images

The gatsby-image docs cover this in some depth as well https://www.gatsbyjs.org/packages/gatsby-image/

**Comment by ooloth** [comment]
Thanks for the replies!

Just to confirm, @KyleAMathews' answer is different than @Undistraction's, and "set the maxWidth to the actual maxWidth" would mean 1500px in the example above?

**Comment by Undistraction** [comment]
Sorry. Ignore me. Was trying to help based on my own experience but I'm obviously wide of the mark.

**Comment by ooloth** [comment]
Not at all! We were thinking the exact same thing. 👍

**Comment by KyleAMathews** [comment]
Haha — yeah, responsive images can be super confusing. The hope of the graphql schemas is that you can think about things in pretty straightforward ways and then Gatsby does the hard work in generating all the necessary thumbnails and setting them up for you.

So just think pixels and forget about resolutions. Browsers are really good now at figuring out what images to grab as long as you give them options and enough info.
